src/app/api/query/auth.ts

```diff
@@ -1,91 +1,89 @@
-import {
-  useMutation,
-  useQueryClient,
-  type UseQueryOptions,
-} from "@tanstack/react-query";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
 
 import { useWebsocketAwareQuery } from "@/app/api/query/base";
+import {
+  mutationOptionsWithHeaders,
+  queryOptionsWithHeaders,
+} from "@/app/api/utils";
 import type {
   CompleteIntroData,
-  CompleteIntroError,
-  CompleteIntroResponse,
+  CompleteIntroErrors,
+  CompleteIntroResponses,
   CreateOauthProviderData,
-  CreateOauthProviderError,
-  CreateOauthProviderResponse,
+  CreateOauthProviderErrors,
+  CreateOauthProviderResponses,
   DeleteOauthProviderData,
-  DeleteOauthProviderError,
-  DeleteOauthProviderResponse,
+  DeleteOauthProviderErrors,
+  DeleteOauthProviderResponses,
   GetMeWithSummaryData,
-  GetMeWithSummaryError,
-  GetMeWithSummaryResponse,
+  GetMeWithSummaryErrors,
+  GetMeWithSummaryResponses,
   GetOauthProviderData,
-  GetOauthProviderError,
-  GetOauthProviderResponse,
+  GetOauthProviderErrors,
+  GetOauthProviderResponses,
   LoginData,
-  LoginError,
-  LoginResponse,
+  LoginErrors,
+  LoginResponses,
   Options,
   UpdateOauthProviderData,
-  UpdateOauthProviderError,
-  UpdateOauthProviderResponse,
+  UpdateOauthProviderErrors,
+  UpdateOauthProviderResponses,
+} from "@/app/apiclient";
+import {
+  deleteOauthProvider,
+  updateOauthProvider,
+  createOauthProvider,
+  completeIntro,
+  getOauthProvider,
+  getMeWithSummary,
+  login,
 } from "@/app/apiclient";
 import {
-  completeIntroMutation,
-  createOauthProviderMutation,
-  deleteOauthProviderMutation,
-  getMeWithSummaryOptions,
   getMeWithSummaryQueryKey,
-  getOauthProviderOptions,
   getOauthProviderQueryKey,
-  loginMutation,
-  updateOauthProviderMutation,
 } from "@/app/apiclient/@tanstack/react-query.gen";
 
 export const useAuthenticate = (mutationOptions?: Options<LoginData>) => {
-  return useMutation<LoginResponse, LoginError, Options<LoginData>>({
-    ...loginMutation(mutationOptions),
+  return useMutation({
+    ...mutationOptionsWithHeaders<LoginResponses, LoginErrors, LoginData>(
+      mutationOptions,
+      login
+    ),
   });
 };
 
-export const useGetCurrentUser = (
-  options?: Options<GetMeWithSummaryData>,
-  queryOptions?: UseQueryOptions<
-    GetMeWithSummaryData,
-    GetMeWithSummaryError,
-    GetMeWithSummaryResponse
-  >
-) => {
+export const useGetCurrentUser = (options?: Options<GetMeWithSummaryData>) => {
   return useWebsocketAwareQuery({
-    ...(getMeWithSummaryOptions(options) as UseQueryOptions<
-      GetMeWithSummaryData,
-      GetMeWithSummaryError,
-      GetMeWithSummaryResponse
-    >),
-    ...queryOptions,
+    ...queryOptionsWithHeaders<
+      GetMeWithSummaryResponses,
+      GetMeWithSummaryErrors,
+      GetMeWithSummaryData
+    >(options, getMeWithSummary, getMeWithSummaryQueryKey(options)),
+    retry: false, // explicitly set retry to false
```

Comment:
I also don't know how I feel about this - while we don't want this to retry on app load (because it increases load times when not logged in), we probably *would* want it to retry in places like the preferences form. Is this overridable?